### PR TITLE
BUG: fix NaN in ND linear interpolation outside convex hull

### LINKED ISSUE #926: BUG: N-D Function with linear interpolation sometimes returns NaN or raises errors for valid-looking queries
### Describe the bug
When using `Function(..., interpolation="linear")` with N-D input data, nan values or error can be raised depending on the data set:

- One query returns `nan` silently.
- Another query raises an exception.

This happens in in this example:

```python
from rocketpy import Function

data = [
    [0.0, 0.0, 0.000],
    [0.0, 0.1, 0.1],
    [0.0, 0.4, 0.4],
    [0.3, 0.2, 0.15],
]
f = Function(data, interpolation="linear")
print(f(0.3, 0.3))   # returns nan

aa = Function([[1, 1, 10], [2, 2, 20], [3, 3, 30]], interpolation="linear")
print(aa(4, 4))       # raises error
```

- bb(0.3, 0.3) returns nan without explicit warning/error.
- aa(4, 4) raises an exception.

This appears to happen only with very specific point configurations/datasets.
Please investigate the underlying cause and clarify intended behavior for these edge cases.

### PR DESCRIPTION
**Pull request by YuriCastroDev**
## Pull request type

- [x] Code changes (bugfix, features)
- [x] Code maintenance (refactoring, formatting, tests)

## Checklist

- [x] Tests for the changes have been added (if needed)
- [x] Lint (`black rocketpy/ tests/`) has passed locally 
- [x] All tests (`pytest tests -m slow --runslow`) have passed locally
- [x] `CHANGELOG.md` has been updated (if relevant)

## Current behavior

Closes #926.

When using `Function(..., interpolation="linear")` with N-D data,
querying a point inside the axis-aligned bounding box but outside
the convex hull silently returns `nan` with no warning.

```python
data = [
    [0.0, 0.0, 0.000],
    [0.0, 0.1, 0.100],
    [0.0, 0.4, 0.400],
    [0.3, 0.2, 0.150],
]
f = Function(data, interpolation="linear")
print(f(0.3, 0.3))  # nan
```

## New behavior

`f(0.3, 0.3)` no longer returns `nan`. Points outside the convex
hull are correctly identified and routed to the extrapolation method.

## Breaking change

- [ ] Yes
- [x] No

## Additional information

`__get_value_opt_nd` used per-axis min/max (bounding box) to decide
between interpolation and extrapolation. `LinearNDInterpolator`
operates on the Delaunay triangulation (convex hull), not the
bounding box — points in the gap between the two were silently
sent to the interpolator and came back as `nan`.

Fix: compute `Delaunay(self._domain)` once at setup, pass it to
`LinearNDInterpolator` (reuses the triangulation at no extra cost),
and store it as `self._nd_triangulation`. In `__get_value_opt_nd`,
replace the bounding box check with `find_simplex(args) < 0`.
Shepard, RBF and other ND interpolators are unaffected.

### CODE REVIEW

**Review by Gui-FernandesBR — APPROVED** [review]
LGTM

File: rocketpy/mathutils/function.py [hunk 1]
```diff
@@ -827,8 +830,11 @@ def __get_value_opt_nd(self, *args):
         min_domain = self._domain.T.min(axis=1)
         max_domain = self._domain.T.max(axis=1)
 
-        lower, upper = args < min_domain, args > max_domain
-        extrap = np.logical_or(lower.any(axis=1), upper.any(axis=1))
+        if hasattr(self, "_nd_triangulation"):
+            extrap = self._nd_triangulation.find_simplex(args) < 0
+        else:
+            lower, upper = args < min_domain, args > max_domain
+            extrap = np.logical_or(lower.any(axis=1), upper.any(axis=1))
```
[the remark this answers is not in the document]
Comment (YuriCastroDev) [reply rocketpy/mathutils/function.py]: Good catch! The `else` branch doesn't cover this case since `_nd_triangulation` 
persists after `set_interpolation()` is called with a different method. 
Accepting the suggestion.

**Review by phmbressan — APPROVED** [review]
Good implementation. Much nicer than my initial idea of computing the interpolation and, with a mask, forwarding any existing `NaN`s to extrapolate. This is more mathematically sound and likely faster.

About the linter issue, it seems a false positive so I would simply disable that line:

```python
from scipy.spatial import Delaunay  # pylint: disable=no-name-in-module
```

**Comment by YuriCastroDev** [comment]
I'm available to keep contributing,  if you have any open issues or tasks that need help, I'd appreciate any guidance on where to focus next!
